### PR TITLE
Typical service water heating module

### PR DESCRIPTION
**Pull request by mdahlhausen**
Pull request overview
---------------------
This PR replaces the `model_add_typical_swh` with `create_typical_service_water_heating`. It no longer requires standards or comstock data attached to a space type to add service hot water heating assumptions. It also adds water equipment to individual spaces, rather than by space type.

### Pull Request Author
 - [x] Method changes or additions
 - [x] Data changes or additions
 - [x] Added tests for added methods
 - [x] If methods have been deprecated, update rest of code to use the new methods
 - [x] Documented new methods using [yard syntax](https://rubydoc.info/gems/yard/file/docs/GettingStarted.md)
 - [x] Resolved yard documentation errors for new code (ran `bundle exec rake doc`)
 - [x] Resolved rubocop syntax errors for new code (ran `bundle exec rake rubocop`)
 - [x] All new and existing tests passes

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [x] Perform a code review on GitHub
 - [x] All related changes have been implemented: method additions, changes, tests
 - [x] Check rubocop errors
 - [x] Check yard doc errors
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If a new feature, test the new feature and try creative ways to break it
 - [x] CI status: all green or justified

### CODE REVIEW

File: lib/openstudio-standards/service_water_heating/create_typical.rb [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+                                                                                                service_water_pump_head: 0.01,
+                                                                                                service_water_pump_motor_efficiency: 1.0,
+                                                                                                water_heater_capacity: water_heater_capacity_w,
+                                                                                                water_heater_volume: water_heater_volume_m3,
+                                                                                                water_heater_fuel: dedicated_water_heating_fuel,
+                                                                                                number_of_water_heaters: num_water_heaters,
+                                                                                                add_piping_losses: true,
+                                                                                                floor_area: OpenStudio.convert(space.floorArea * space.multiplier, 'ft^2', 'm^2').get,
```
Comment (eringold) [inline lib/openstudio-standards/service_water_heating/create_typical.rb:165]: third time doing this OpenStudio.convert - consider putting into a variable at the start of the `model.getSpaces.each` loop

**Comment by mdahlhausen** [comment]
**Adding some comments for future reference:**

This PR removes 'template' and 'service water heating area' from the service water heating data.

The new schema looks like this:
```
{
  space_type: 'name' ,
  building_types: 'bldgA',
  water_use_equipment: [
    {
      equipment_name: 'dishwasher',
      peak_flow_rate: 130.0,
      peak_flow_rate_per_area: nil,
      loop_type: 'Shared',
      temperature: 140.0,
      flow_rate_schedule: 'SchA',
      sensible_fraction: 0.2,
      latent_fraction: 0.05
    },
  ]
}
```

Here are other fields that might be added in the future:
  - minimum tank size (if one per space)
  - water heating fuel (if one per space)
  - minimum space size (only apply if a space is at least this large)
  - check adjaceny to other spaces (combine water uses together into one use if adjacent spaces are the same space type)

File: lib/openstudio-standards/service_water_heating/information.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        water_use_equip_sch = water_use_equip_sch.get
+        if water_use_equip_sch.to_ScheduleRuleset.is_initialized
+          water_use_equip_sch = water_use_equip_sch.to_ScheduleRuleset.get
+          max_sch_value = OpenstudioStandards::Schedules.schedule_ruleset_get_min_max(water_use_equip_sch)['max']
+        elsif water_use_equip_sch.to_ScheduleConstant.is_initialized
+          water_use_equip_sch = water_use_equip_sch.to_ScheduleConstant.get
+          max_sch_value = OpenstudioStandards::Schedules.schedule_constant_get_min_max(water_use_equip_sch)['max']
+        elsif water_use_equip_sch.to_ScheduleCompact.is_initialized
+          water_use_equip_sch = water_use_equip_sch.to_ScheduleCompact.get
+          max_sch_value = OpenstudioStandards::Schedules.schedule_compact_get_min_max(water_use_equip_sch)['max']
```
Comment (eringold) [inline lib/openstudio-standards/service_water_heating/information.rb:42]: why not just [OpenstudioStandards::Schedules.schedule_get_min_max(water_use_equip_sch)['max']](https://github.com/NREL/openstudio-standards/blob/e5ef38ac30dcbc1a3eefa3901fa3438588452f5a/lib/openstudio-standards/schedules/information.rb#L15)

File: lib/openstudio-standards/service_water_heating/information.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        # Get peak flow rate from water use equipment definition
+        peak_flow_rate_m3_per_s = water_use_equip.waterUseEquipmentDefinition.peakFlowRate
+
+        # Calculate adjusted flow rate based on the peak fraction found in the flow rate fraction schedule
+        adjusted_peak_flow_rate_m3_per_s = max_sch_value * peak_flow_rate_m3_per_s
+        adjusted_max_flow_rates_gal_per_hr << OpenStudio.convert(adjusted_peak_flow_rate_m3_per_s, 'm^3/s', 'gal/hr').get
+      end
+
+      # Sum gph values from water use equipment to use in formula
+      total_adjusted_flow_rate_gal_per_hr = adjusted_max_flow_rates_gal_per_hr.inject(:+)
```
Comment (eringold) [inline lib/openstudio-standards/service_water_heating/information.rb:57]: I think `Array::sum` is better

**Review by eringold — APPROVED** [review]
just minor comments

File: lib/openstudio-standards/service_water_heating/create_typical.rb (hunk 1 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/service_water_heating/create_typical.rb:165]: there is actually a unit conversion error here too. fixed.

File: lib/openstudio-standards/service_water_heating/information.rb (hunk 2 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/service_water_heating/information.rb:42]: that is a better choice. changed.

File: lib/openstudio-standards/service_water_heating/information.rb (hunk 3 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/service_water_heating/information.rb:57]: changed

**Pull request merged**